### PR TITLE
ft(UTAPI-23): Limit account level credentials

### PR DESCRIPTION
**Pull request by tmacro**

### CODE REVIEW

**Review by mrepop — APPROVED** [review]

File: libV2/vault/client.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    /**
+    * Returns canonical Ids for a given list of account Ids
+    * @param {string[]} accountIds - list of account ids
+    * @param {object} log - Werelogs request logger
+    * @return {Promise} -
+    */
+    getCanonicalIds(accountIds, log) {
+        log.debug('retrieving canonical ids for account ids', {
```
Comment (jonathan-gramain) [inline libV2/vault/client.js:31]: it might be useful to log the account IDs as well here

File: libV2/vault/client.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+                    resolve({ authed: true, authInfo, authorizedResources });
+                }, 's3', [policyContext],
+            );
+        });
+    }
+
+    getUsersById(userIds, log) {
+        log.debug('retrieving user arns for user ids', {
+            method: 'Vault.getUsersById',
```
Comment (jonathan-gramain) [inline libV2/vault/client.js:88]: same, logging the user IDs can help

File: libV2/vault/client.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                }));
+    }
+}
+
+const vault = new Vault(config);
+auth.setHandler(vault);
+
+module.exports = {
+    Vault,
+    vault,
```
Comment (jonathan-gramain) [inline libV2/vault/client.js:112]: This looks confusing to me as the class and instance only differ by case, maybe rename to `vaultInstance`? But it's a refactor and not part of this PR's changes so you can skip.

File: libV2/vault/index.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
+const assert = require('assert');
+const { vault } = require('./client');
+const metadata = require('../metadata');
+const errors = require('../errors');
+
+// Will be used for user keys
+// eslint-disable-next-line no-unused-vars
```
Comment (jonathan-gramain) [inline libV2/vault/index.js:7]: This exception may not be needed

File: libV2/vault/index.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    // Accounts are only allowed access if they are the owner of the bucket
+    case 'buckets': {
+        let buckets;
+        try {
+            buckets = await Promise.all(
+                resources.map(bucket => metadata.getBucket(bucket)),
+            );
+        } catch (error) {
+            log.error('failed to fetch metadata for bucket', { error });
```
Comment (jonathan-gramain) [inline libV2/vault/index.js]: It could be more precise to log right after `metadata.getBucket` returns, to log which bucket got an error

File: tests/functional/v2/server/testListMetrics.js [hunk 6]
```diff
@@ -81,140 +83,104 @@ function opsToResp(operations) {
...
+            });
+        });
 
-        const [accountMetric] = accountResp.body;
-        assert.deepStrictEqual(accountMetric.storageUtilized, [0, 0]);
-        assert.deepStrictEqual(accountMetric.numberOfObjects, [0, 0]);
-        assert.deepStrictEqual(accountMetric.incomingBytes, 0);
-        assert.deepStrictEqual(accountMetric.outgoingBytes, 0);
-    });
+        it('should not list metrics for an different account', async () => {
```
Comment (jonathan-gramain) [inline tests/functional/v2/server/testListMetrics.js]:
```suggestion
        it('should return AccessDenied for a different account', async () => {
```

**Review by jonathan-gramain — APPROVED** [review]

File: libV2/vault/client.js (hunk 3 above)
Comment (tmacro) [reply libV2/vault/client.js:112]: I'll rename it to `VaultWrapper`

File: libV2/vault/index.js (hunk 4 above)
Comment (tmacro) [reply libV2/vault/index.js:7]: It's cleaned up in the service user PR
